deepchecks/tabular/context.py

```diff
@@ -30,16 +33,118 @@
 ]
 
 
+class _DummyModel:
+    """Contains all the data + properties the user has passed to a check/suite, and validates it seamlessly.
+
+    Parameters
+    ----------
+    train: Dataset
+        Dataset, representing data an estimator was fitted on
+    test: Dataset
+        Dataset, representing data an estimator predicts on
+    y_pred_train: np.ndarray
+        Array of the model prediction over the train dataset.
+    y_pred_test: np.ndarray
+        Array of the model prediction over the test dataset.
+    y_proba_train: np.ndarray
+        Array of the model prediction probabilities over the train dataset.
+    y_proba_test: np.ndarray
+        Array of the model prediction probabilities over the test dataset.
+    """
+
+    train: pd.DataFrame
+    test: pd.DataFrame
+    ind_map: t.Dict[str, t.Dict[t.Any, int]]
+    y_pred: t.Dict[str, np.ndarray]
+    y_proba: t.Dict[str, np.ndarray]
+
+    def __init__(self,
+                 train: Dataset,
+                 test: Dataset,
+                 y_pred_train: np.ndarray,
+                 y_pred_test: np.ndarray,
+                 y_proba_train: np.ndarray,
+                 y_proba_test: np.ndarray,):
+        self.ind_map = defaultdict(dict)
+
+        if train is not None and test is not None:
+            # check if datasets have same indexes
+            if set(train.data.index) & set(test.data.index):
+                train.data.index = map(lambda x: f'train-{x}', list(train.data.index))
+                test.data.index = map(lambda x: f'test-{x}', list(test.data.index))
+                warnings.warn('train and test datasets have common index - adding "train"/"test"'
+                              ' prefixes. To avoid that provide datasets with no common indexes '
+                              'or don\'t use the static predictions feature.')
```

Comment:
As a user, I won't know what a "static prediction" is

Reply:
Can you offer another name?

Reply:
"Or pass the model object instead of the predictions"?